### ISSUE TITLE
small target detection

### ISSUE THREAD

**Issue by yeyuyanzhulei** [ultralytics/ultralytics#22164]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am currently working on a small object detection task. Previously, I trained on small objects using the yolov8.pt and yolov8-p2p.yaml configuration files. I also looked for small object detection projects from other authors, but only found one infrared small object recognition. Moreover, it did not provide good assistance in the use of this task, and was even worse than the small object detection configuration file that comes with YOLO. The size of the image is also quite large, approximately 2600×1900. At present, the detection can be successful, but there are cases of false detection and missed detection (the main reason is the presence of disturbances). Do any experts have any methods to improve stability?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yeyuyanzhulei, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get moving quickly—an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (e.g., small object detection), please share:
-Exact Ultralytics version and environment (OS, CUDA, Python, PyTorch)
-Your model and config used (e.g., P2-enabled config), full training/validation command, and key hyperparameters
-Sample images with labels (small-object crops and full-resolution examples ~2600×1900)
-Training logs/metrics (precision/recall/mAP, losses), and inference settings (img size, conf, IoU/NMS)
-Any augmentations, mosaics, or tiling/cropping steps you’ve tried

Also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). These steps often help reduce false positives/negatives and improve stability on small targets.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can try training a P2 model

https://github.com/Y-T-G/community/blob/main/cfg/detect/yolo11-p2.yaml

**Comment by knoppmyth** [comment]
You can also look at [SAHI](https://docs.ultralytics.com/guides/sahi-tiled-inference/).

**Comment by glenn-jocher** [comment]
Good call on SAHI—given 2600×1900 frames and tiny targets, I’d recommend switching to Ultralytics YOLO11, training with a P2 head (as suggested), and/or running sliced inference; a solid starting point is 512×512 tiles with 20–30% overlap while tuning `conf`/`iou` to balance FP/FN, and training-time tiling can also help as noted in our [Image tiling for small objects](https://docs.ultralytics.com/guides/model-evaluation-insights/). 

```python
from sahi import AutoDetectionModel
from sahi.predict import get_sliced_prediction

detection_model = AutoDetectionModel.from_pretrained(
    model_type="ultralytics", model_path="yolo11n.pt", confidence_threshold=0.25, device="cuda:0"
)

result = get_sliced_prediction(
    "path/to/2600x1900.jpg", detection_model,
    slice_height=512, slice_width=512, overlap_height_ratio=0.2, overlap_width_ratio=0.2
)
```

**Comment by yeyuyanzhulei** [comment]
Thank you all for your suggestions. I'll try to train a P2 model. Regarding the use of the sahi library, I have incorporated it into the detection code, but it seems to cause a delay situation? Will this cause problems in real-time detection?

**Comment by yeyuyanzhulei** [comment]
> You can try training a P2 model
> 
> https://github.com/Y-T-G/community/blob/main/cfg/detect/yolo11-p2.yaml

Thank you for your suggestion. Do I still need to make certain changes to this configuration file based on the pixel of the images I trained?

**Comment by Y-T-G** [comment]
No, you can use it directly

**Comment by thi23459112** [comment]
> You can try training a P2 model
> 
> https://github.com/Y-T-G/community/blob/main/cfg/detect/yolo11-p2.yaml

I’d like to ask, when using P2 training, how do we specify different model versions?
For example, the traditional training command is:

yolo task=detect mode=train model=yolo11.yaml data=myvoc.yaml epochs=30


If we switch to P2 training, should the command be:

model=yolo11-p2.yaml


or:

model=yolo11s-p2.yaml

**Comment by Y-T-G** [comment]
You just add it to the name while loading. Don't modify the original filename.

`model=yolo11s-p2.yaml`
